src/excel/Workbook.js

```diff
@@ -47,6 +48,27 @@ export class Workbook extends NamedItemContainer {
     return new Table(this._oneDrive, `${this._uri}/tables`, name, this._log);
   }
 
+  async addTable(address, hasHeaders, name) {
+    if (name) {
+      const names = await this.getTableNames();
+      if (names.includes(name)) {
+        throw new StatusCodeError(`Table name already exists: ${name}`, 409);
+      }
+    }
+    const result = await this._oneDrive.doFetch(`${this.uri}/tables/add`, false, {
+      method: 'POST',
+      body: {
+        address,
+        hasHeaders,
+      },
+    });
+    const table = this.table(result.name);
+    if (name && name !== table.name) {
+      await table.rename(name);
+    }
```

Comment:
if name is given and it doesn't equal the automatically generated one, do a `rename`